pytorch_lightning/trainer/trainer.py

```diff
@@ -2022,7 +2022,10 @@ def num_processes(self) -> int:
 
     @property
     def root_gpu(self) -> Optional[int]:
-        return self._accelerator_connector.root_gpu
+        rank_zero_deprecation(
+            "Method `trainer.root_gpu` is deprecated please use `trainer.strategy.root_device`", stacklevel=5
+        )
+        return self._strategy.root_device
```

Comment:
This used to return the index, so we need to keep that behaviour

```suggestion
        return self._strategy.root_device.index
```